src/main/java/gregtech/common/metatileentities/storage/MetaTileEntityCreativeChest.java

```diff
@@ -0,0 +1,172 @@
+package gregtech.common.metatileentities.storage;
+
+import codechicken.lib.render.CCRenderState;
+import codechicken.lib.render.pipeline.IVertexOperation;
+import codechicken.lib.vec.Cuboid6;
+import codechicken.lib.vec.Matrix4;
+import com.google.common.collect.Lists;
+import gregtech.api.GTValues;
+import gregtech.api.capability.GregtechCapabilities;
+import gregtech.api.capability.IEnergyContainer;
+import gregtech.api.gui.GuiTextures;
+import gregtech.api.gui.ModularUI;
+import gregtech.api.gui.widgets.*;
+import gregtech.api.metatileentity.MetaTileEntity;
+import gregtech.api.metatileentity.MetaTileEntityHolder;
+import gregtech.client.renderer.texture.Textures;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.common.util.Constants;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.items.CapabilityItemHandler;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.items.ItemHandlerHelper;
+import net.minecraftforge.items.ItemStackHandler;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.List;
+import java.util.function.Function;
+
+public class MetaTileEntityCreativeChest extends MetaTileEntity {
+
+    private int itemsPerCycle = 1;
+    private int ticksPerCycle = 1;
+    private final ItemStackHandler handler = new ItemStackHandler(1);
```

Comment:
What do you think about overriding `getSlotLimit` for this `ItemStackHandler` to return 1, just for a nicer display when adding something to the slot. Right now if you click in a stack of 4 of an item, the slot will display the stack of 4, which might be a bit confusing considering that the amount produced is set by the field entry and not the stacksize in the slot.